### PR TITLE
fix: Change non-existent secrets dir into a warning

### LINKED ISSUE #2175: secrets_dir configuration setting raises an error when the directory doesn't exist
### Checks

* [x] I added a descriptive title to this issue
* [x] I have searched (google, github) for similar issues and couldn't find anything
* [x] I have read and followed [the docs](https://pydantic-docs.helpmanual.io/) and still think this is a bug

# Bug

Output of `python -c "import pydantic.utils; print(pydantic.utils.version_info())"`:
```
             pydantic version: 1.7.3
            pydantic compiled: True
                 install path: ~/.pyenv/versions/3.8.5/envs/fetcher/lib/python3.8/site-packages/pydantic
               python version: 3.8.5 (default, Sep  4 2020, 14:39:18)  [GCC 9.3.0]
                     platform: Linux-5.8.0-7630-generic-x86_64-with-glibc2.29
     optional deps. installed: ['typing-extensions', 'devtools']
```

I have declared a settings model that can read values from the environment, a `.env` file, or from Docker secrets as follows.

```py
import pydantic

class Settings(BaseSettings):

    password: SecretStr = Field(..., env="APP_PASSWORD")

    class Config:
        """Configure settings values discovery locations."""

        env_file = ".env"
        secrets_dir = "/run/secrets"
```

[As per the documentation](https://pydantic-docs.helpmanual.io/usage/settings/#field-value-priority), checking for a value from a secret is the fourth attempt pydantic makes. However, when I just work in my development environment where the directory `/run/secrets` doesn't exist, instantiating the model raises an error.

```
pydantic.env_settings.SettingsError: directory "/run/secrets" does not exist
```

This is annoying because it happens even though I have the values in a `.env` file. Thus I have to circumvent this by using `Settings(_secrets_dir=".")` which would mean that I need a special case for my dev environment. I think, when values are provided to the instance, or from the environment, or from `.env` first, it shouldn't matter whether the `secrets_dir` exists or not.

### PR DESCRIPTION
**Pull request by davidolrik**
## Change Summary

Previously a non existent secrets directory would throw an error, this has now been turned into a warning.

This makes local development easier, as a secrets directory is usually something that only exists in Kubernetes or Docker.

## Related issue number

Fixes #2175

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [X] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

**Review by PrettyWood — COMMENTED** [review]
LGTM if this is the desired behaviour.
Note that it will conflict with https://github.com/samuelcolvin/pydantic/pull/2107

File: pydantic/env_settings.py [hunk 1]
```diff
@@ -61,8 +61,10 @@ def _build_secrets_files(self, _secrets_dir: Union[Path, str, None] = None) -> D
         secrets_path = Path(secrets_dir).expanduser()
 
         if not secrets_path.exists():
-            raise SettingsError(f'directory "{secrets_path}" does not exist')
-        if not secrets_path.is_dir():
+            warnings.warn(f'directory "{secrets_path}" does not exist')
+            return secrets
+
+        if secrets_path.exists() and not secrets_path.is_dir():
```
Comment (PrettyWood) [inline pydantic/env_settings.py:67]:
```suggestion
        if not secrets_path.is_dir():
```

File: changes/2175-davidolrik.md [hunk 2]
```diff
@@ -0,0 +1 @@
+fix: Change non-existent secrets dir from exception to warning
```
Comment (PrettyWood) [inline changes/2175-davidolrik.md:1]:
```suggestion
Allow non-existent secrets directories by issuing only a warning
```

Comment (davidolrik) [reply changes/2175-davidolrik.md:1]:
```suggestion
Allow non-existent secrets directory by only issuing a warning
```

**Comment by samuelcolvin** [comment]
thanks so much. I think this makes a lot of sense.

**Pull request merged**